### PR TITLE
Split tests for reshaping in two files

### PR DESCRIPTION
**Pull request by etiennebacher**
Makes it easier to distinguish tests for `data_to_long()` and those for `data_to_wide()`. Also adds a few tests where missing

### CODE REVIEW

File: R/data_to_long.R [hunk 1]
```diff
@@ -116,6 +116,11 @@ data_to_long <- function(data,
     }
   }
 
+  # nothing to select?
+  if (!length(cols)) {
```
Comment (IndrajeetPatil) [inline R/data_to_long.R:120]:
```suggestion
  if (length(cols) > 0L) {
```

File: R/data_to_long.R [hunk 2]
```diff
@@ -116,6 +116,11 @@ data_to_long <- function(data,
...
+    insight::format_error("No columns found for reshaping data.")
+  }
+
   if (length(names_to) > 1 && is.null(names_sep) && is.null(names_pattern)) {
```
Comment (IndrajeetPatil) [inline R/data_to_long.R:124]:
```suggestion
  if (length(names_to) > 1L && is.null(names_sep) && is.null(names_pattern)) {
```

**Review by IndrajeetPatil — COMMENTED** [review]
SGTM. Thanks!

**Comment by IndrajeetPatil** [comment]
@etiennebacher Something seems to have broken. Can you have a look?

**Comment by etiennebacher** [comment]
Yes I'll fix it

**Pull request merged**